apis/definition.yaml

```diff
@@ -0,0 +1,126 @@
+                            version:
+                              description: flux helm-chart version to run.
+                              type: string
+                              default: "2.10.6"
+                          required:
+                            - version
+                        flux-sync:
+                          description: Configuration for the Flux Sync Helm-Chart.
+                          type: object
+                          properties:
+                            version:
+                              description: flux sync helm-chart version to run.
+                              type: string
+                              default: "1.7.2"
+                          required:
+                            - version
+                    source:
+                      type: object
+                      properties:
+                        git:
+                          type: object
+                          properties:
+                            interval:
+                              default: "5m0s"
+                              description: Interval at which the GitRepository URL is checked for
+                                updates. This interval is approximate and may be subject to jitter
+                                to ensure efficient use of resources.
+                              pattern: ^([0-9]+(\.[0-9]+)?(ms|s|m|h))+$
+                              type: string
+                            timeout:
+                              default: "60s"
+                              description: Timeout for Git operations like cloning, defaults to
+                                60s.
+                              pattern: ^([0-9]+(\.[0-9]+)?(ms|s|m))+$
+                              type: string
+                            url:
+                              description: URL specifies the Git repository URL, it can be an HTTP/S
+                                or SSH address.
+                              pattern: ^(http|https|ssh)://.*$
+                              type: string
+                            path:
+                              type: string
+                              default: "/"
+                            ref:
+                              description: Reference specifies the Git reference to resolve and 
+                                monitor for changes.
+                              type: object
+                              properties:
+                                name:
+                                  description: "Name of the reference to check out; takes precedence
+                                    over Branch, Tag and SemVer. \n It must be a valid Git reference:
+                                    https://git-scm.com/docs/git-check-ref-format#_description Examples:
+                                    \"refs/heads/main\", \"refs/tags/v0.1.0\", \"refs/pull/420/head\",
+                                    \"refs/merge-requests/1/head\""
```

Comment:
why do we escape special characters here? is it optimized for some view?

Reply:
Yes better display in the marketplace